files/usr/lib/jeos-firstboot

```diff
@@ -201,15 +204,14 @@ if [ -z "$JEOS_LOCALE" ]; then
     fi
 
     if [ -n "$newlocale" ]; then
-        locale="${newlocale}.UTF-8"
-        systemd_firstboot_args+=("--locale=$locale")
+        JEOS_LOCALE="${newlocale}.UTF-8"
+        systemd_firstboot_args+=("--locale=$JEOS_LOCALE")
 
         # Run langset to get consolefont and keymap set up
-        run langset.sh $locale || warn $"Setting the locale failed"
+        run langset.sh $JEOS_LOCALE || warn $"Setting the locale failed"
     fi
 else
-    locale="$JEOS_LOCALE"
-    systemd_firstboot_args+=("--locale=$locale")
+    systemd_firstboot_args+=("--locale=$JEOS_LOCALE")
```

Comment:
Should be moved below the fi. The `[ -n "$newlocale" ]` case is never true anyway.

Reply:
How is so? `$newlocale` have always a default value

Reply:
Sorry, I meant false.

Reply:
Right! In that case removing the `else` will add two `--locale=$JEOS_LOCALE` into the systemd array. I can think that this will not break anything, but is confusing.

Reply:
I meant something like this:

```
if [ -z "$JEOS_LOCALE" ]; then
    default="en_US"
    [ -f /etc/locale.conf ] && locale_lang="$(awk -F= '$1 == "LANG" { split($2,fs,"."); print fs[1]; exit }' /etc/locale.conf)"
    [ -n "$locale_lang" ] && default="$locale_lang"

    list=() # Set by findlocales
    newlocale="$default"
    if ! findlocales; then
        d --msgbox $"No locales found" 0 0
    elif [ "${#list[@]}" -eq 2 ]; then
        newlocale="${list[0]}"
        d --msgbox $"Locale set to $newlocale.\nTo change to a different one, install glibc-locale and use\n'localectl set-locale LANG=ex_AMPLE.UTF-8'." 8 50
    else
        d --default-item "$default" --menu $"Select System Locale" 0 0 $dh_menu "${list[@]}"
        newlocale="${result}"
    fi

    JEOS_LOCALE="${newlocale}.UTF-8"
fi

run langset.sh $JEOS_LOCALE || warn $"Setting the locale failed"
systemd_firstboot_args+=("--locale=$JEOS_LOCALE")
```

Reply:
Very well. I wonder if a similar refactor can be done in the keytable section